c7n/resources/cw.py

```diff
@@ -833,3 +833,47 @@ def process(self, resources):
                     client.disassociate_kms_key(logGroupName=r['logGroupName'])
             except client.exceptions.ResourceNotFoundException:
                 continue
+
+
+@LogGroup.action_registry.register('put-subscription-filter')
+class SubscriptionFilter(BaseAction):
+    """Create/Update a subscription filter and associate with a log group
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: cloudwatch-put-subscription-filter
+            resource: log-group
+            actions:
+              - type: put-subscription-filter
+                filter_name: AllLambda
+                filter_pattern: ip
+                destination_arn: arn:aws:logs:us-east-1:1234567890:destination:lambda
+                distribution: Random
+    """
+    schema = type_schema(
+        'put-subscription-filter',
+        filter_name={'type': 'string'},
```

Comment:
@raghavendarreddyg i believe some of these fields are mandatory. If so, update the schema to enforce that.

Reply:
Added `validate()`  to check the required fields

Reply:
@raghavendarreddyg the convention is to do as much validation at the schema definition level as possible rather than using the validate method. Schema definition will show up in doc vs validation will not.

Do something like this
```
    schema = type_schema(
        'put-subscription-filter',
        filter_name={'type': 'string'},
        filter_pattern={'type': 'string'},
        destination_arn={'type': 'string'},
        distribution={'enum': ['Random', 'ByLogStream']},
        required=['filter_name', 'destination_arn']
   )
```
I didn't required `filter_pattern` because I don't think that is required right? If it's empty, then the subscription filter is for all the log messages. Right?

Reply:
@darrendao `filterPattern` is required by the boto3 function, `put_subscription_filter()`. The function at least needs the value to be empty string. It blows up otherwise. One option is to add it to the required fields and enforce the user to pass the empty string in the policy in order to filter all the log messages. The second option is to change `filter_pattern = self.data.get('filter_pattern')` to `filter_pattern = self.data.get('filter_pattern', '')` in the `process()` method to pass the empty string to boto3 function if the policy is missing a value. What is the preferred way?

Reply:
@raghavendarreddyg `filter_pattern = self.data.get('filter_pattern', '')` seems better to me

Reply:
@darrendao Pushed the change